pkg/kubelet/kubelet.go

```diff
@@ -2436,6 +2442,47 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}
 }
 
+// CheckpointContainer tries to checkpoint a container. The parameters are used to
+// look up the specified container. If the container specified by the given parameters
+// cannot be found an error is returned. If the container is found the container
+// engine will be asked to checkpoint the given container into the kubelet's default
+// checkpoint directory.
+func (kl *Kubelet) CheckpointContainer(
+	podUID types.UID,
+	podFullName,
+	containerName string,
+	options *runtimeapi.CheckpointContainerRequest,
+) error {
+	container, err := kl.findContainer(podFullName, podUID, containerName)
+	if err != nil {
+		return err
+	}
+	if container == nil {
+		return fmt.Errorf("container %v not found", containerName)
+	}
+
+	if strings.HasPrefix(options.Location, "/") {
+		return fmt.Errorf("archive_location cannot start with '/' but %q was provided", options.Location)
+	}
+
+	if !strings.Contains(options.Location, "/") && options.Location != "" {
+		options.Location = filepath.Join(kl.getCheckpointsDir(), options.Location)
+	}
```

Comment:
suggest limiting params to the web service to.. just container id and possibly timeout if others agree timeout should be an option

Reply:
Filtered out `..`. Over the time developing this PR the option to store the checkpoint as an image in the local registry has become the preferred use case as it makes restoring via create/start possible without an additional CRI RPC call for restore. It also makes it easy to transfer the checkpoint image between Kubernetes Nodes by just pushing the image to registry.

containerd already supports the checkpoint image in my corresponding PR (https://github.com/containerd/containerd/pull/6965) and my CRI-O PR will also support it. I would prefer to keep it the way it is now. Storing the image in a registry provides the option to restore containers in Kubernetes without any additional Kubernetes changes. It can be handled transparently in the underlying CRI implementation.

Reply:
After discussing this on slack I removed the parameter to specify a custom location and the PR does now only what was agreed upon in the corresponding KEP.